enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,264 @@
+- Finally, if the MSC will attempt to patch the `MachineSet` if required. Failure to do so will cause a degrade. 
+- Any other failures in the above steps will report an error; but degrades will only be in the specific cases mentioned above. Certain failures may also be as a result of an unsupported architecture or an unsupported platform. This is necessary because support for platforms will be phased in(and some platforms may not even desire this support)
+
+#### Degrade Mechanism
+
+One possible strategy would be for the MSC to degrade the worker `MachineConfigPool` via a new [MachineConfigPoolConditionType](https://github.com/openshift/api/blob/master/machineconfiguration/v1/types.go#L492). This would be an API change, but a fairly simple one is it only adding a new type. The node controller(another sub controller within the MCC) would then [check for this condition](https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/status.go#L142C34-L142C34) and degrade the worker pool, effectively degrading the operator.
+
+As mentioned in the above section, degrading will only happen in two scenarios:
+- Translating the ignition stub to spec 3 fails. This is likely more fatal and won't get fixed without the editing the ignition stub manually.
+- Patching of the MachineSet fails. This is likely due to a temporary API server outage and will resolve itself without user intervention.
+
+Every degrade will be associated with a machineset. As a result, the MSC will have to maintain a local list of currently "degraded" machinesets, and remove/add to this list during a sync loop as necessary. Based on the changes to this list, the MSC can then intiate, update or clear a degrade condition on the operator.
+
+#### Reverting to original bootimage
+
+Few strategies here:
+
+##### via Annotations
+
+The MSC will maintain a backup of the the following via annotation to the `MachineSet`:
+- `io.openshift.mco-managed-factory-image=` storing the original provider image reference prior to when the feature was turned on
+- `io.openshift.mco-managed-factory-secret=` storing the original stub secret name
+- `io.openshift.mco-managed-last-image=` storing the last provider image reference before an update
+- `io.openshift.mco-managed-last-secret=` storing the last stub secret name before an update
+
+##### via ConfigMap
+
+The MSC will maintain a backup of the following per machinset, with these keys:
+- `$(machine_set_name)-factory-image` storing the original provider image reference prior to when the feature was turned on
+- `$(machine_set_name)-factory-secret` storing the original stub secret name prior to the feature was turned on
+- `$(machine_set_name)-last-image` storing the last provider image reference before an update
+- `$(machine_set_name)-last-secret` storing the last stub secret name before an update
```

Comment:
I think you should aim for a concrete API rather than annotations or a configmap

Reply:
Will drop these. To be clear, you're open to adding new fields to a MachineSet object's status? (: Or does something else make sense?

Reply:
Ah, sorry just saw the other comments where you talked about this - github hid them from me :( I'll respond to them shortly.